### PR TITLE
Add json output format for ddt command

### PR DESCRIPTION
**Pull request by therealak12**
Currently `ddt` can only print output in human readable format. This PR adds the option to print found vulnerabilities only in JSON format.

This pull request makes the following changes:
* Adds `-o, --output-format` flag for `ddt` command

cc @bhamail / @DarthHater

### CODE REVIEW

**Comment by bhamail** [comment]
@therealak12 Thanks for this PR. I noticed our CI was not configured to build from forks. I've fixed that, but then noticed the build failing for this PR. Could you take a look and see if you can fix that failure? 

FWIW, you can likely reproduce the CI error locally by following these steps: [CI Debug Notes](https://github.com/sonatype-nexus-community/jake/blob/main/.circleci/circleci-readme.md)

I was able to reproduce the error locally using these steps.

**Comment by therealak12** [comment]
@bhamail I fixed the CI errors. Would you please check again?

**Comment by therealak12** [comment]
@bhamail You may need to uncheck ci/circleci. See [here](https://circleci.com/docs/2.0/workflows/#workflows-waiting-for-status-in-github).

**Comment by bhamail** [comment]
@therealak12 It looks like CI is behaving now. Might need to resolve the merge conflicts GH is flagging.

File: jake/__main__.py [hunk 1]
```diff
@@ -197,21 +201,32 @@ def sbom(verbose, quiet, conda, targets, output):
...
   Example usage:\n
       Python scan: jake ddt\n
       Conda scan: conda list | jake ddt -c\n
   """
-  __banner(quiet)
   __setup_logger(verbose)
   __check_stdin(conda)
 
+  if output_format == 'json':
+    __toggle_stdout(on=False)
```
Comment (bhamail) [inline jake/__main__.py:219]: Not a comment on your changes, just noticing our use of `__toggle_stdout` is getting a bit hard to follow.

This change seems to work sanely in my local testing.

**Review by bhamail — COMMENTED** [review]
Looks good to me. 

Holding off on a +1 until I better understand the changes from PR #55.

File: jake/__main__.py (hunk 1 above)
Comment (therealak12) [reply jake/__main__.py:219]: Yeah :)))
I was trying to find a nice way to only print the json output and I suddenly noticed this boolean! Not a good way but it's solving the problem for now.

**Comment by bhamail** [comment]
@therealak12 Could you merge the recent changes from `main` into this branch/PR? I'm guessing the branch conflicts will be resolved by doing so.

After that, looks good to go.

**Comment by therealak12** [comment]
Sure I will.

File: jake/pip/pip.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ def __init__(self, targets=None, requirements_file_path=None):
     self.requirements_file_path = requirements_file_path
     if targets:
       self._working_set = pkg_resources.WorkingSet(ast.literal_eval(targets))
-    self._coords = self.generate_dependencies()
+    self.__coords = self.generate_dependencies()
```
Comment (bhamail) [inline jake/pip/pip.py:34]: Did this sneak back in?

**Review by bhamail — CHANGES_REQUESTED** [review]
I think maybe one change was unintended, otherwise ready to go.

File: jake/pip/pip.py (hunk 2 above)
Comment (therealak12) [reply jake/pip/pip.py:34]: Would you please correct it by yourself?
If i do, the CI will need another commit from you in order to run.

Comment (bhamail) [reply jake/pip/pip.py:34]: Perfect. I didn't want to muck with your PR without consent. Duck and cover, here it comes.

**Review by bhamail — APPROVED** [review]
LGTM

**Pull request merged**